### PR TITLE
Flatten list values

### PR DESCRIPTION
**Pull request by mariarivera30**
Now Info is flatten before it goes to the csv row formatter func.

### CODE REVIEW

File: pybossa/exporter/task_csv_export.py [hunk 1]
```diff
@@ -148,11 +149,44 @@ def _format_csv_row(self, row, headers):
     def _handle_row(self, writer, t, headers):
         writer.writerow(self._format_csv_row(self.merge_objects(t),
                                              headers=headers))
+
+    def _flatten_json_iterative(self,dictionary):
```
Comment (jayflo) [inline pybossa/exporter/task_csv_export.py]: You might be able to simplify this a bit:

```
def flatten(key_value_pairs, key_prefix='', return_value=None):
    return_value = return_value if return_value is not None else {}
    for k, v in key_value_pairs:
        key = '{}_{}'.format(prefix, k)
        if isinstance(v, dict):
            flatten(v.iteritems(), key, return_value)
        elif isinstance(v, list):
            flatten(enumerate(v), key, return_value)
        else:
            return_value[key] = return_value
    return return_value
```

But yours might have some different behavior.

File: pybossa/exporter/task_csv_export.py [hunk 2]
```diff
@@ -148,11 +149,44 @@ def _format_csv_row(self, row, headers):
...
+
+        return dictionary
+
     def _get_csv_with_filters(self, out, writer, table, project_id,
                               expanded, filters):
         objs = browse_tasks_export(table, project_id, expanded, filters)
         rows = [obj for obj in objs]
-
+        for row in rows:
+            if (row['info']):
```
Comment (fiorda) [inline pybossa/exporter/task_csv_export.py]: parentheses are not needed

File: pybossa/exporter/task_csv_export.py (hunk 1 above)
Comment (fiorda) [reply pybossa/exporter/task_csv_export.py]: Looks good, use `six.iteritems`

File: pybossa/exporter/task_csv_export.py [hunk 3]
```diff
@@ -148,11 +149,44 @@ def _format_csv_row(self, row, headers):
...
+                    new_key = u'{}{}{}'.format(parent_key, '_', key) 
+                    yield new_key, value
+            elif isinstance(parent_value, list):
+                for i, value in enumerate(parent_value, start=1): 
+                    new_key = u'{}{}{}'.format(parent_key, '_', str(i)) 
+                    yield new_key, value
+            else:
+                yield parent_key, parent_value    
+                
+        while True:
```
Comment (dchhabda) [inline pybossa/exporter/task_csv_export.py]: can this function decorated with timeout; something similar to https://pypi.org/project/timeout-decorator/?

File: pybossa/exporter/task_csv_export.py (hunk 1 above)
Comment (dchhabda) [inline pybossa/exporter/task_csv_export.py]: cosmetic; add space after comma in func parameters

Comment (mariarivera30) [reply pybossa/exporter/task_csv_export.py]: Yes, will do it this way. thanks

Comment (jayflo) [reply pybossa/exporter/task_csv_export.py]: The iterator is nice.  This should work:

```python
def flatten(key_value_pairs, key_prefix='', return_value=None):
    return_value = return_value if return_value is not None else {}
    for k, v in key_value_pairs:
        key = k if not key_prefix else '{}_{}'.format(key_prefix, k)
        if isinstance(v, dict):
            iterator = flatten(v.items(), key, return_value)
        elif isinstance(v, list):
            iterator = flatten(enumerate(v), key, return_value)
        else:
            iterator = [(key, v)]
        for kk, vv, in iterator:
            yield kk, vv


print(dict(flatten({
    'a': [0, 1],
    'b': {
        'c': 1
    },
    'd': 1
}.items())))
```

**Pull request merged**